File: CareConnect/CareConnect.Models/SearchObjects/AppointmentSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class AppointmentSearchObject : BaseSearchObject<AppointmentAdditionalData>
    {
        public string? AppointmentType { get; set; } 

        public DateTime? DateGTE { get; set; }

        public DateTime? DateLTE { get; set; }

        public DateTime? Date { get; set; }

        public string? AttendanceStatusName { get; set; }

        public string? EmployeeFirstName { get; set; }

        public string? EmployeeLastName { get; set; } 

        public string? StartTime { get; set; }

        public string? EndTime { get; set; }

        public string? Status { get; set; }

        public string? ChildFirstName { get; set; } 

        public string? ChildLastName { get; set; }

        public int? ClientId { get; set; }

        public int? ChildId { get; set; }

        public string? ClientUsername { get; set; }

        public int? ServiceTypeId { get; set; }

        public string? ServiceNameGTE { get; set; }

        public int? EmployeeAvailabilityId { get; set; }

    }
}

File: UI/careconnect_mobile/lib/models/search_objects/appointment_search_object.dart
import 'package:careconnect_mobile/models/search_objects/appointment_additional_data.dart';
import 'package:json_annotation/json_annotation.dart';

part 'appointment_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class AppointmentSearchObject {
  String? fts;
  String? appointmentType;
  DateTime? dateGTE;
  DateTime? dateLTE;
  String? attendanceStatusName;
  String? employeeFirstName;
  String? employeeLastName;
  String? startTime;
  String? endTime;
  String? status;
  String? childFirstName;
  String? childLastName;
  int? clientId;
  int? childId;
  String? clientUsername;
  int? serviceTypeId;
  String? serviceNameGTE;
  int? employeeAvailabilityId;
  DateTime? date;
  AppointmentAdditionalData? additionalData;
  int? page;
  String? sortBy;
  bool? sortAscending;
  bool? includeTotalCount;

  AppointmentSearchObject({
    this.fts,
    this.appointmentType,
    this.dateGTE,
    this.dateLTE,
    this.attendanceStatusName,
    this.employeeFirstName,
    this.employeeLastName,
    this.startTime,
    this.endTime,
    this.status,
    this.childFirstName,
    this.childLastName,
    this.clientId,
    this.childId,
    this.clientUsername,
    this.serviceTypeId,
    this.serviceNameGTE,
    this.employeeAvailabilityId,
    this.date,
    this.additionalData,
    this.page,
    this.sortAscending,
    this.includeTotalCount,
    this.sortBy,
  });

  factory AppointmentSearchObject.fromJson(Map<String, dynamic> json) =>
      _$AppointmentSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentSearchObjectToJson(this);
}
